### PR TITLE
[qna] switch to customizations

### LINKED ISSUE #23505: [QnA] Refactor to use new patch customizations
The current generated QnA SDK has manual edits to add method overloads to the Mixin.
The updated autorest support for patching operations should be used instead to allow for easy regeneration.

The WebPubSub updates should provide a good template: #23504 
FYI @iscai-msft

### PR DESCRIPTION
**Pull request by iscai-msft**
fixes #23505

### CODE REVIEW

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/_version.py [hunk 1]
```diff
@@ -6,4 +6,4 @@
 # Changes may cause incorrect behavior and will be lost if the code is regenerated.
 # --------------------------------------------------------------------------
 
-VERSION = "1.1.0b1"
+VERSION = "1.0.0"
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/_version.py:9]: oops - we're going backwards ;)

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/aio/_patch.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+Follow our quickstart for examples: https://aka.ms/azsdk/python/dpcodegen/python/customize
+"""
+from typing import Any, List
+from azure.core.credentials import AzureKeyCredential
+from ._question_answering_client import QuestionAnsweringClient as QuestionAnsweringClientGenerated
+
+
+class QuestionAnsweringClient(QuestionAnsweringClientGenerated):
+    __doc__ = QuestionAnsweringClientGenerated.__doc__
```
Comment (annatisch) [inline sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/aio/_patch.py:15]: Don't we need to update the docstring with the `default_language` kwarg?

File: sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/_version.py (hunk 1 above)
Comment (iscai-msft) [reply sdk/cognitivelanguage/azure-ai-language-questionanswering/azure/ai/language/questionanswering/_version.py:9]: oh no, updating!

**Review by annatisch — APPROVED** [review]

**Pull request merged**